### PR TITLE
Create your own Enumerable

### PR DESCRIPTION
**Pull request by fabgrel10**
In this PR:

- [x] Create a class MyList that has an instance variable @list.
- [x] In MyList implement a method #each that yields successive members of @list and uses the MyEnumerable module.
- [x] Create a module My[Enumerable](https://ruby-doc.org/core-3.0.0/Enumerable.html) that implements the methods all?, any? and filter.
- [x] Each class and module should has a separate .rb file.

### CODE REVIEW

File: my_enumerable.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  #
+  #   any? { |n| 5 >= n}
+  #   # => true
+  #
+  # Returns true if the condition is met.
+  def any?
+    flag = false
+    each { |item| flag = true if yield(item) }
+    flag
+  end
```
Comment (Emmyn5600) [inline my_enumerable.rb:33]: ### Optional suggestion:

- [ ] Your implementation is working however I would recommend checking on the following implementation too.

`All`

```
def my_all?(param = nil)
    if block_given?
      to_a.my_each { |n| return false unless yield(n) }
    elsif param.nil?
      to_a.my_each { |n| return false if !n || n.nil? }
    elsif !param.nil? && (param.is_a? Class)
      to_a.my_each { |n| return false unless [n.class, n.class.superclass, n.class.superclass].include?(param) }
    elsif !param.nil? && (param.is_a? Regexp)
      to_a.my_each { |n| return false unless n.match(param) }
    else
      to_a.my_each { |n| return false unless n == param }
    end
    true
  end
```

`Any`

```
  def my_any?(param = nil)
    if block_given?
      to_a.my_each { |n| return true if yield(n) }
    elsif param.nil?
      to_a.my_each { |n| return true if n }
    elsif !param.nil? && (param.is_a? Class)
      to_a.my_each { |n| return true if [n.class, n.class.superclass, n.class.superclass].include?(param) }
    elsif !param.nil? && (param.is_a? Regexp)
      to_a.my_each { |n| return true if n.match(param) }
    else
      to_a.my_each { |n| return true if n == param }
    end
    false
  end
```

File: my_list.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+class MyList
+  include MyEnumerable
+
+  def initialize(*list)
+    @list = list
+  end
+
+  def each
+    @list.each { |x| yield(x) if block_given? }
+  end
```
Comment (Emmyn5600) [inline my_list.rb:13]: - [ ] Let us add a simple loop like `while loop` here in this method to be sure if all created MyEnumerables are working properly.

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @fabgrel10 , @Leboroz and @andres-condezo 

Good job so far! I know you have worked hard on making this milestone however, There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight :champagne:

- All linters are passing :heavy_check_mark:
- Good PR description :heavy_check_mark:


## Required Changes :recycle:

- [ ]  Check the comment under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!:clap::clap::clap:

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by fabgrel10** [comment]
All the suggestions and the required changes were implemented. Thank you for the review @Emmyn5600. Definitely the implementations of all? and any? are more robust.

**Review by EricMbouwe — APPROVED** [review]
### Hi @fabgrel10, 

# Status: Approved ✔️
Excellent job on this project. Keep it up 👏
Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by fabgrel10** [comment]
Thank you for the review @EricMbouwe!

**Pull request merged**